rero_ils/modules/documents/api.py

```diff
@@ -207,17 +207,20 @@ def index_contributions(self, bulk=False):
             ref = contribution['agent'].get('$ref')
             if ref:
                 contrib, online = Contribution.get_record_by_ref(ref)
-            cont_pid = contribution['agent'].get('pid')
-            if cont_pid:
-                contrib = Contribution.get_record_by_pid(cont_pid)
+            else:
+                cont_pid = contribution['agent'].get('pid')
+                if cont_pid:
+                    contrib = Contribution.get_record_by_pid(cont_pid)
             if contrib:
                 if bulk:
                     contributions_ids.append(contrib.id)
                 else:
                     contrib.reindex()
         if contributions_ids:
-            IlsRecordsIndexer().bulk_index(
-                contributions_ids, doc_type=['cont'])
+            from ..contributions.api import ContributionsIndexer
+            indexer = ContributionsIndexer()
+            indexer.bulk_index(contributions_ids)
+            indexer.process_bulk_queue()
```

Comment:
After some discuss with @lauren-d this afternoon, this syntax seems synchronous ; if we're right, there is no benefits using this syntax. But calling `process_bulk_queue.apply_async()` from `rero_ils.modules.tasks`, the process become async (but you need that Celery is up and running).

Will test again tomorrow using this syntax

Reply:
I think you are right. I have one problem with asynchronous indexing. Sometimes the indexing is not done and I have not yet found out why.

Reply:
Rewritten.